projects/plugins/social/src/js/hooks/use-product-info/index.js

```diff
@@ -0,0 +1,46 @@
+import apiFetch from '@wordpress/api-fetch';
+import { useCallback, useEffect, useState } from 'react';
+
+const getPriceData = productObject => {
+	return {
+		price: productObject.cost / 12,
+		introOffer: productObject.introductory_offer
+			? productObject.introductory_offer.cost_per_interval / 12
+			: null,
+	};
+};
+
+const parsePromotedProductInfo = response => {
+	const currencyCode = response.advanced.currency_code;
+	return {
+		currencyCode,
+		basic: getPriceData( response.basic ),
+		advanced: getPriceData( response.advanced ),
+	};
+};
+
+/**
+ * Hook to retrieve the product info for the pricing page.
+ *
+ * @returns {object} - The product info containing the currency and the plan prices.
+ */
+export default function useProductInfo() {
+	const [ productInfo, setProductInfo ] = useState( null );
+
+	const getAsyncInfo = useCallback( async () => {
+		try {
+			const socialPromotedProductInfo = await apiFetch( {
+				path: '/jetpack/v4/social-promoted-product-info',
+			} );
+			setProductInfo( parsePromotedProductInfo( socialPromotedProductInfo ) );
+		} catch {
+			setProductInfo( null );
+		}
+	}, [] );
+
+	useEffect( () => {
+		getAsyncInfo();
+	}, [ getAsyncInfo ] );
```

Comment:
Did you create `getAsyncInfo` to be able to use async/await inside the `useEffect`? If so, you could also define the whole function inside the `useEffect` callback and call it right after defining it, or wrap it in an IIFE.

Something like: 

```js
useEffect(() => {
	(async () => {
		try {
			const socialPromotedProductInfo = await apiFetch( {
				path: '/jetpack/v4/social-promoted-product-info',
			} );
			setProductInfo( parsePromotedProductInfo( socialPromotedProductInfo ) );
		} catch {
			setProductInfo( null );
		}
	})();
}, []);
```

But again, it's mostly personal preference 😅

Reply:
Yeah, I just like the way it looks, to me it looks cleaner, I was never a fan of IIFEs, but yeah as you said it's a matter of preference 😄 I can change it if you want.

Reply:
No all good, feel free to leave it! Ideally we should have an ESLint rule that enforces one way or another to improve consistency.